sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -359,6 +359,21 @@ trait StateStoreProvider {
   /** Return an instance of [[StateStore]] representing state data of the given version */
   def getStore(version: Long): StateStore
 
+  /**
+   * This is an optional method, used by snapshotStartBatchId option when reading state generated
```

Comment:
Why do we treat this as optional ? we are adding implementations for both store providers right ?

Reply:
Yes both built-in providers implement this method. However, if user wants to create their own state store provider, this method should not be required.

Reply:
I feel like its ok to require providers to implement this

cc - @HeartSaVioR - thoughts ?

Reply:
I will create a new optional trait and make the two providers implement it.